internal/ingress/controller/store/store.go

```diff
@@ -301,6 +310,51 @@ func (s k8sStore) GetServiceEndpoints(key string) (*corev1.Endpoints, error) {
 	return s.listers.Endpoint.ByKey(key)
 }
 
+func (s *k8sStore) annotationToNodeLabelObj(nodeLabelAnnotation string) *[]NodeLabel {
+	var nodeLabels []NodeLabel
+	cleaned := strings.Replace(nodeLabelAnnotation, " ", "", -1)
+	cleaned = strings.Replace(cleaned, ",", " ", -1)
+	valSlice := strings.Fields(cleaned)
+
+	for _, val := range valSlice {
+		s := strings.Split(val, "=")
+		nodeLabels = append(nodeLabels, NodeLabel{
+			Key:   s[0],
+			Value: s[1],
+		})
+	}
+
+	return &nodeLabels
+}
+
+func (s *k8sStore) compareNodeLabels(node *corev1.Node, nodeLabelsObj *[]NodeLabel) (bool, error) {
+	for _, nodeLabel := range *nodeLabelsObj {
+		val, ok := node.ObjectMeta.Labels[nodeLabel.Key]
+		if !ok {
+			continue
+		}
+
+		if val == nodeLabel.Value {
```

Comment:
the current logic imposes that the nodeSelector is in OR condition(e.g. one label match is selected), which is counterintuitive and doesn't match kubernetes's label selector's behavior(AND condition)